PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -313,6 +335,42 @@ function loadProperties(localSettingsFile : string) {
     }
 }
 
+function getDtpBaseUrl(settings : ReadOnlyProperties) : string {
+    const dtpUrl = settings['dtp.url'];
+    if (!isNullOrWhitespace(dtpUrl)) {
+        try {
+            return new URL(dtpUrl).href;
+        } catch (err) {
+            tl.error('Invalid dtp.url value in local settings file.');
+            return '';
+        }
+    }
+
+    const dtpServer = settings['dtp.server'];
+    if (isNullOrWhitespace(dtpServer)) {
+        tl.warning('Both dtp.url and dtp.server properties are not specified in local settings file.');
+        return '';
+    }
+
+    let dtpBaseUrl : URL;
+    try {
+        dtpBaseUrl = new URL('https://' + dtpServer);
```

Comment:
It's possible to have invalid port number (not within range) and invalid character in path segment, correct?
We may want to double check for those two properties, too.